cni/cmd/istio-cni-taint/main.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A simple daemonset binary to repair pods that are crashlooping
+// after winning a race condition against istio-cni
+
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	client "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+
+	"istio.io/istio/cni/pkg/taint"
+	"istio.io/pkg/log"
+)
+
+type ControllerOptions struct {
+	RunAsDaemon  bool           `json:"run_as_daemon"`
+	TaintOptions *taint.Options `json:"taint_options"`
+}
+
+const (
+	LeassLockName      = "istio-taint-lock"
+	LeaseLockNamespace = "kube-system"
+)
+
+var (
+	loggingOptions = log.DefaultOptions()
+
+	rootCmd = &cobra.Command{
+		Use:          "taint controller",
+		Short:        "taint controller command line interface",
+		SilenceUsage: true,
+		Long: `Istio CNI taint controller used for monitoring on the readiness of configmap defined
+Critical labels, this can be run as a standalone command line tool or as a daemon.
+If it run as command line tool, it will check the readiness of critical labels defined in configmap
+if some critical labels are not ready, taint the corresponding node. otherwise, it will run as a
+kubernetes controller and checking on the readiness of critical label
+`,
+		PersistentPreRunE: configureLogging,
```

Comment:
call `
	// Attach the Istio logging options to the command.
	loggingOptions.AttachCobraFlags(rootCmd)
`

otherwise there is no way to configure logging